Chat App/ViewModel/Pub/PubRegisterViewModel.swift

```diff
@@ -0,0 +1,154 @@
+//
+//  PubRegisterViewModel.swift
+//  Chat App
+//
+//  Created by Amiel Jireh Cordova on 12/14/20.
+//  Copyright © 2020 Cordova, Jireh. All rights reserved.
+//
+
+import Foundation
+import RxCocoa
+import RxSwift
+
+protocol RegisterViewModelInputs {
+    func verifyUsernameAvailability(userInput: String)
+    func verifyUserInput(userInput: String) -> Bool
+    func verifyPasswordMatch(userInput: String!, comparable: String!) -> Bool
+    func registerUser(submittedUsername: String!, password: String!)
+}
+
+protocol RegisterViewModelOutputs {
+    var alertTitle: String { get }
+    var alertMessage: String { get }
+    var alertActionLabel: String { get }
+    var shouldShowAlert: PublishSubject<Bool> { get }
+    var shouldShowLoading: PublishSubject<Bool> { get }
+    var isUsernameAvailable: PublishSubject<Bool> { get }
+    var shouldProceedtoServer: PublishSubject<Bool> { get }
+}
+
+protocol RegisterViewModelTypes {
+    var inputs: RegisterViewModelInputs { get }
+    var outputs: RegisterViewModelOutputs { get }
+}
+
+class PubRegisterViewModel: RegisterViewModelInputs, RegisterViewModelOutputs, RegisterViewModelTypes {
+    var shouldShowAlert = PublishSubject<Bool>()
+    var shouldShowLoading = PublishSubject<Bool>()
+    var isUsernameAvailable = PublishSubject<Bool>()
+    var shouldProceedtoServer = PublishSubject<Bool>()
+    var inputs: RegisterViewModelInputs { return self }
+    var outputs: RegisterViewModelOutputs { return self }
+    
+    var alertTitle = ""
```

Comment:
maybe copy thte format from  the outputs protocol so instead of direct initialization why not put a data type also var alertTitle: String = ""

Reply:
copied the protocol format for this property.

Reply:
Adjusted variable initialization to match protocol.